File: session6-addnlds/Main.py
# inheritance
class Animal:
    def sound(self):
        print("Animal makes a sound")


class Dog(Animal):
    def sound(self):
        print("Dog barks")


def main():
    # animal = Animal()
    # animal.sound()

    dog = Dog()
    dog.sound()


if __name__ == "__main__":
    main()

File: session6-addnlds/Main.js
class Animal {

    constructor(name) {
        this.name = name;
    }

    sound() {
        console.log('Animal makes a sound');
    }
}

class Dog extends Animal {

    constructor(name) {
        super(name);
    }

    sound() {
        super.sound();
        console.log('Dog barks');
    }
}

let dog = new Dog('Dog');
dog.sound(); // Dog barks
print(dog.name); // Dog